app/src/main/java/com/adyen/nexoapp/terminal/SelectedTerminalInfo.kt

```diff
@@ -0,0 +1,55 @@
+package com.adyen.nexoapp.terminal
+
+import java.util.*
+
+// todo object can eventually retain some previous selected terminals to quickly select one on UI
+
+object SelectedTerminalInfo {
+
+    private var _terminalModelName: String = ""
+    private var _serialNumber: String = ""
+    private val _serialNumberLength: Int = 9
+    private var _ipAddress: String = ""
+
+    val poiId: String
+        get() {
+            val trimmedSerialNumber = _serialNumber.trim()
+
+            if (trimmedSerialNumber.length != _serialNumberLength) {
+                throw IllegalArgumentException("Invalid serial number length")
+            }
+
+            if (_terminalModelName.isEmpty()) {
+                throw IllegalArgumentException("No terminal model specified")
+            }
+
+            return "$_terminalModelName-$trimmedSerialNumber"
+        }
+
+    var terminalModelName: String
```

Comment:
There's no need to declare both `_terminalModelName` and `terminalModelName`. You can handle this in the same property. That also applies for the other fields.

Additionally, throwing `IllegalArgumentException`s here is prone to cause crashes.

Reply:
I will have a look